src/DaPigGuy/PiggyShopUI/commands/ShopCommand.php

```diff
@@ -107,57 +103,71 @@ public function showCategoryItems(Player $player, ShopCategory $category): void
 
     public function showItemPage(Player $player, ShopCategory $category, ShopItem $item): void
     {
-        $form = new CustomForm(function (Player $player, ?array $data) use ($category, $item): void {
-            if ($data !== null) {
-                if (!is_numeric($data[1]) || (int)$data[1] < 0) {
-                    $player->sendMessage($this->plugin->getMessage("menu.item.numeric"));
-                    return;
-                }
-                if (!$item->canSell() || !$data[2]) {
-                    if ($this->plugin->getEconomyProvider()->getMoney($player) < $item->getBuyPrice() * (int)$data[1]) {
-                        $player->sendMessage($this->plugin->getMessage("buy.not-enough-money", ["{PRICE}" => $item->getBuyPrice() * (int)$data[1], "{DIFFERENCE}" => $item->getBuyPrice() * (int)$data[1] - $this->plugin->getEconomyProvider()->getMoney($player)]));
-                        return;
-                    }
-                    $purchasedItem = clone $item->getItem();
-                    $purchasedItem->setCount($purchasedItem->getCount() * (int)$data[1]);
-                    if (!$player->getInventory()->canAddItem($purchasedItem)) {
-                        $player->sendMessage($this->plugin->getMessage("buy.not-enough-space"));
+        $this->plugin->getEconomyProvider()->getMoney($player, function (float|int $amount) use ($player, $category, $item): void {
```

Comment:
checking money twice? nothing should've changed right https://github.com/DaPigGuy/PiggyShopUI/pull/40/files#diff-8dc91ca2542ab7de9b25a9de2ff11c44c3bb73baf0e229d0b10df80e37c81574R114